### PR TITLE
fix wp list progress text in percent done is cut by scroll

### PR DESCRIPTION
**Pull request by 0xF013**
[`* `#16651` wp list progress text in percent done is cut by scroll`](http://www.openproject.org/work_packages/16651)

### CODE REVIEW

**Comment by TeatroIO** [comment]
I've prepared a stage. <a href='http://0xf013-feature-16651-wp-list-progress-text-in-percent-done-is-c.opf-openproject-2af22bf12c7dfdad4ddf.ttrcloud.com' target='_blank'>Click to open.</a>

**Comment by tessi** [comment]
I'm afraid this is more a work-around than a fix. You add a 20px padding (which "steals" 20px from the screen). Now, the table seems to be 20px too short on the right side:

selection_058

I think, the original error comes because of the re-rendering of the table header (I heard the table header width is re-calculated depending on the table contents). You can see that (without your change) when you add the `% done` column and then any other column. The table width fits perfectly.

If you remove the last column (so that the `% done` column is the last one), and then resize the browser window randomly and keep doing so for a while, you can observe how the table header recalculates its width until it is too small.

I suggest to fix the re-calculation logic, instead of adding the 20px border for all columns.

**Comment by 0xF013** [comment]
@tessi I found that the resize script doesn't account for the case where you add and then remove an element that makes the table switch from 100% width to a fixed width, so the page remains horizontally scrollable. I will fix that and add 16-20px to the table width if it exceeds the window viewport width

**Comment by myabc** [comment]
@0xF013 great. I hadn't considered that scenario. That would probably go a long way to fixing some of the remaining buggy behaviour.

File: app/assets/javascripts/angular/work_packages/directives/work-packages-table-directive.js [hunk 1]
```diff
@@ -101,8 +108,25 @@ angular.module('openproject.workPackages.directives')
         });
       }
 
+      function invalidateWidths() {
+        var table          = element.find('table'),
+            innerContainer = element.find('.work-packages-table--container-inner'),
+            backgrounds    = element.find('.work-packages-table--header-background,' +
```
Comment (myabc) [inline app/assets/javascripts/angular/work_packages/directives/work-packages-table-directive.js]: @0xF013 You should be able to move most of the variable declarations/initialisations into the main directive function body. With the exception of the footer (whose DOM node may or may not be present), the value of these selectors will not change.

**Comment by 0xF013** [comment]
@myabc amended

**Comment by myabc** [comment]
@0xF013 :+1: awesome. looks good. unfortunately, you may need to rebase again.

**Comment by 0xF013** [comment]
@myabc After rebase it freezes in IE for some seconds. I am installing parallels to test it properly since virtualbox sucks

**Pull request merged**